### PR TITLE
feat(connection-form): support system CA store loading COMPASS-4105

### PR DESCRIPTION
**Pull request by addaleax**
### Checklist
- [ ] New tests and/or benchmarks are included
- [ ] Documentation is changed or added

## Motivation and Context

- [ ] Bugfix
- [x] New feature
- [ ] Dependency update
- [ ] Misc

## Types of changes

- [ ] *Backport Needed*
- [ ] Patch (non-breaking change which fixes an issue)
- [x] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/connection-form/src/components/advanced-options-tabs/tls-ssl-tab/tls-certificate-authority.tsx [hunk 1]
```diff
@@ -26,20 +38,38 @@ function TLSCertificateAuthority({
...
+          updateCAFile(null, event.target.checked);
+        }}
+        id="useSystemCA-input"
+        data-testid="useSystemCA-input"
+        label="Use System Certificate Authority"
+        disabled={disabled}
+        checked={useSystemCA}
+        bold={false}
+      />
+      <Description
```
Comment (mcasimir) [inline packages/connection-form/src/components/advanced-options-tabs/tls-ssl-tab/tls-certificate-authority.tsx:66]: @Anemy found out that both label and description should be passed as "Label" for checkboxes, should be already changed on main for the rest of the connection form.

File: packages/connection-form/src/components/advanced-options-tabs/tls-ssl-tab/tls-certificate-authority.tsx [hunk 2]
```diff
@@ -26,20 +38,38 @@ function TLSCertificateAuthority({
     <FormFieldContainer className={caFieldsContainer}>
       <FileInput
         description={'Learn More'}
-        disabled={disabled}
+        disabled={disabled || useSystemCA}
```
Comment (mcasimir) [inline packages/connection-form/src/components/advanced-options-tabs/tls-ssl-tab/tls-certificate-authority.tsx:41]: Just a question about this, are those strict alternatives? or is theoretically possible to pass part of the CA chain as a file and have other certs for parent CAs in a system folder? Nothing to address anyway, I'm just curious.

**Review by mcasimir — APPROVED** [review]
👍

File: packages/connection-form/src/components/advanced-options-tabs/tls-ssl-tab/tls-certificate-authority.tsx (hunk 2 above)
Comment (addaleax) [reply packages/connection-form/src/components/advanced-options-tabs/tls-ssl-tab/tls-certificate-authority.tsx:41]: As soon as one certificate in the chain is part of the CA, the server’s certificate will be considered trusted – it wouldn’t really make sense to specify two separate lists of trusted CAs for a single endpoint.

Additionally, the way that the Node.js driver currently works is that it reads the `tlsCAFile` and stores it in the `ca` option which `useSystemCA` maps to, so they would conflict in the Node.js driver options parsing. (Plus, we should keep the liberty to eventually implement `useSystemCA` via `tlsCAFile` ourselves as well.)

File: packages/connection-form/src/components/advanced-options-tabs/tls-ssl-tab/tls-certificate-authority.tsx (hunk 1 above)
Comment (addaleax) [reply packages/connection-form/src/components/advanced-options-tabs/tls-ssl-tab/tls-certificate-authority.tsx]: Ah thanks for pointing that out, I missed that – updated to match those changes :)

**Comment by addaleax** [comment]
Alright, in order for the e2e test to work, we’ll need mongosh webpack config changes.

Since I’m OOO for a week, I’ll leave some notes here, in theory this should work – I’ll do as much as I can today, but we’ll see how far we get:

- [ ] Review and merge https://github.com/mongodb-js/mongosh/pull/1221
- [ ] Release mongosh. Our release docs are complete, but because of outstanding build issues, there are some caveats here:
  - [ ] Checkout `main`, run `git pull`
  - [ ] Run `npm run release draft`, confirm everything
  - [ ] Go to evergreen CI, start Slack notification for when the tag build finishes + Slack notification when a build variant in the task fails
  - [ ] **Restart failing tasks**, maybe don’t even wait for a Slack notification but do it every 20 minutes or so – we have a lot of infrastructure flakiness on Windows and macOS hosts (BUILD-12985, BUILD-14243)
  - [ ] Wait until the task is green
  - [ ] Bring https://github.com/mongodb-js/homebrew-core up to date (i.e. merge master from https://github.com/Homebrew/homebrew-core and push to our fork) – this is important, since otherwise publishing homebrew fails (no ticket for this issue afaik)
  - [ ] Run `git pull` again in mongosh (the repo will have updated)
  - [ ] Run `npm run release publish`, confirm everything
  - [ ] Go to evergreen CI, Slack notification for the release tag
  - [ ] Wait for the task to finish – this took almost 7 hours the last time! (BUILD-13863)
  - [ ] Announce in #mongosh, inform Jess or Max about updating the release in JIRA
  - [ ] Notify those involved in https://jira.mongodb.org/browse/HELP-31642 about the release
- [ ] Bump the shell versions all across Compass, open a PR
- [ ] Merge that PR
- [ ] Rebase this PR against main
- [ ] … yay?

File: packages/connection-form/src/components/advanced-options-tabs/tls-ssl-tab/tls-certificate-authority.tsx [hunk 3]
```diff
@@ -26,20 +39,44 @@ function TLSCertificateAuthority({
...
+                [disabledCheckboxDescriptionStyles]: disabled,
+              })}
+            >
+              Use the operating system’s Certificate Authority store.
+            </Description>
+          </>
+        }
+        disabled={disabled}
+        checked={useSystemCA}
+        bold={false}
```
Comment (Anemy) [inline packages/connection-form/src/components/advanced-options-tabs/tls-ssl-tab/tls-certificate-authority.tsx:78]: V small suggestion, looks like `bold={false}`[ is the default](https://www.mongodb.design/component/checkbox/documentation/), so we can probably leave that out
```suggestion
```
I think it exists on one or two of the checkboxes in the connect form which we can probably remove it from too, probably left over from when we were passing `label` as a string.

File: packages/hadron-build/commands/release.js [hunk 4]
```diff
@@ -321,10 +321,6 @@ const installDependencies = util.callbackify(async(CONFIG) => {
  * @api public
  */
 const removeDevelopmentFiles = (CONFIG, done) => {
-  if (CONFIG.platform !== 'darwin') {
-    done();
-    return;
-  }
```
Comment (gribnoysup) [inline packages/hadron-build/commands/release.js:327]: Oh wow

**Review by gribnoysup — APPROVED** [review]

File: packages/compass-e2e-tests/helpers/commands/set-connect-form-state.ts [hunk 5]
```diff
@@ -175,6 +175,9 @@ export async function setConnectFormState(
       Selectors.ConnectionFormTlsAllowInvalidCertificatesCheckbox
     );
   }
+  if (state.useSystemCA) {
```
Comment (lerouxb) [inline packages/compass-e2e-tests/helpers/commands/set-connect-form-state.ts:178]: You probably want to put something in get-connect-form-state as well? maybe update connection-form.test.ts to toggle the field and check the state.

Comment (addaleax) [reply packages/compass-e2e-tests/helpers/commands/set-connect-form-state.ts:178]: Sure, done :+1:

File: packages/connection-form/src/components/advanced-options-tabs/tls-ssl-tab/tls-certificate-authority.tsx (hunk 3 above)
Comment (addaleax) [reply packages/connection-form/src/components/advanced-options-tabs/tls-ssl-tab/tls-certificate-authority.tsx]: Yeah, I just copied it from the other TLS checkboxes but removing it doesn’t appear to make a difference, done :+1:

File: packages/compass-e2e-tests/tests/connection-form.test.ts [hunk 6]
```diff
@@ -6,7 +6,7 @@ import type { Compass } from '../helpers/compass';
 import * as Selectors from '../helpers/selectors';
 import type { ConnectFormState } from '../helpers/connect-form-state';
 
-describe('Connection form', function () {
+describe.only('Connection form', function () {
```
Comment (lerouxb) [inline packages/compass-e2e-tests/tests/connection-form.test.ts:9]: ☝️

**Review by lerouxb — APPROVED** [review]

File: packages/hadron-build/commands/release.js (hunk 4 above)
Comment (Anemy) [reply packages/hadron-build/commands/release.js:327]: Looks like this change took 5+MB off of non-mac releases 🔥

**Pull request merged**